product_warranty/models/product_warranty.py

```diff
@@ -0,0 +1,131 @@
+# ########################################################################
+#                                                                        #
+# Copyright 2015 Vauxoo
+# Copyright (C) 2009-2011  Akretion, Emmanuel Samyn, Benoît Guillot      #
+#                                                                        #
+# This program is free software: you can redistribute it and/or modify   #
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# This program is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+from openerp import api, fields, models
+
+
+class ReturnInstruction(models.Model):
+
+    _name = "return.instruction"
+    _description = "Instructions for product return"
+
+    name = fields.Char('Title', required=True)
+
+    instructions = fields.Text(
+        'Instructions',
+        help="Instructions for product return")
+
+    is_default = fields.Boolean('Is default',
+                                help="If is default, will be use "
+                                "to set the default value in "
+                                "supplier infos. Be careful to "
+                                "have only one default")
+
+
+class ProductSupplierInfo(models.Model):
+
+    _inherit = "product.supplierinfo"
+
+    @api.model
+    def get_warranty_return_partner(self):
+        result = [('company', 'Company'),
+                  ('supplier', 'Supplier'),
+                  ('other', 'Other'),
+                  ]
+        return result
+
+    @api.model
+    def _get_default_instructions(self):
+        """ Get selected lines to add to exchange """
+        instruction_ids = self.env['return.instruction']\
+            .search([('is_default', '=', True)], limit=1)
+        return instruction_ids[0]
+
+    @api.one
+    @api.depends('warranty_return_partner')
+    def _get_warranty_return_address(self):
+        """ Method to return the partner delivery address or if none,
+        the default address
+
+        dedicated_delivery_address stand for the case a new type of
+        address more particularly dedicated to return delivery would be
+        implemented.
+
+        """
+        for record in self:
```

Comment:
If you put @api.one, self is already only one record and you make the operations directly with self.